src/client/mod.rs

```diff
@@ -818,6 +819,49 @@ impl KafkaClient {
         Err(Error::NoHostReachable)
     }
 
+    fn load_api_versions(&mut self) -> Result<()> {
+        let correlation = self.state.next_correlation_id();
+        let now = Instant::now();
+        for broker in self.state.brokers().iter_mut() {
+            debug!("fetch_api_versions: requesting API versions from {}", broker.host());
+            match self.conn_pool.get_conn(broker.host(), now) {
+                Ok(conn) => {
+                    let req = protocol::ApiVersionsRequest::new(correlation,
+                                                                &self.config.client_id);
+                    match __send_request(conn, req) {
+                        Ok(_) => {
+                            match __get_response::<protocol::ApiVersionsResponse>(conn) {
+                                Ok(res) => {
+                                    debug!("fetch_api_versions: got API versions from {}: {:?}",
+                                           conn.host(),
+                                           res);
+
+                                    broker.update_api_versions(res);
+                                }
+                                Err(err) => {
+                                    debug!("fetch_api_versions: failed to get response from {}: {}",
+                                           conn.host(),
+                                           err);
+
+                                    conn.close();
```

Comment:
In fact, Kafka 0.8 will close the connection immediately when it received the unknown `ApiVersionsRequest`, we just mark the connection was closed for the connection pool.